### PR TITLE
openthread-br: bump source to 2024-11-20 and add missing dependencies

### PR DESCRIPTION
**Pull request by itay-sho**
Maintainer: @stintel 
Compile tested: armsr/armv8/generic, and banana pi r4. both aarch64
Run tested: armsr/armv8/generic, and banana pi r4. both aarch64
I tested it until I saw that the service is up (and not crashing) as it was before

Description:
bumped ot-br-posix that openthread-br is based on to latest code.
the major changes required some more compilation flags, and in order to avoid the ot-br agent had to add some dependencies as weell.

dependencies:
1. kmod-tun: otherwise the otbr fails to load and crashed immidiedtly

compilation flags:
1. OTBR_NAT64, OT_NAT64_BORDER_ROUTING: a temporary workaround to a compilation problem with openwrt. there is an issue I've opened about it here: https://github.com/openthread/ot-br-posix/issues/2606
1. OT_TARGET_OPENWRT: wasn't assumed from OTBR_OPENWRT in banana pi r4 (but was in the other target). decided to add it to avoid problem

### CODE REVIEW

**Comment by stintel** [comment]
I'll have to investigate this properly. I cannot believe avahi is needed and I absolutely do not want to add that dependency. Unfortunately I am renovating my house and this will have to wait. To anyone with merge privileges on this repository: do not merge this until further notice.

**Comment by itay-sho** [comment]
> I'll have to investigate this properly. I cannot believe avahi is needed and I absolutely do not want to add that dependency. Unfortunately I am renovating my house and this will have to wait. To anyone with merge privileges on this repository: do not merge this until further notice. 

There is a possibility that adding OT_TARGET_OPENWRT removed avahi client as a mandatory dependency for this. Both were required only on my bpi4 target (and not the other qemu arm target). it did compile a little different after adding it.

Would you like me to check?

**Comment by stintel** [comment]
> > I'll have to investigate this properly. I cannot believe avahi is needed and I absolutely do not want to add that dependency. Unfortunately I am renovating my house and this will have to wait. To anyone with merge privileges on this repository: do not merge this until further notice. 
> 
> There is a possibility that adding OT_TARGET_OPENWRT removed avahi client as a mandatory dependency for this. Both were required only on my bpi4 target (and not the other qemu arm target). it did compile a little different after adding it.
> 
> Would you like me to check?
> 

Please do. I really don't want avahi and dbus as dependency of this package.

**Comment by itay-sho** [comment]
After removing libavahi and enabling OT_TARGET_OPENWRT, the compilation was successful. 
The commit was recreated without libavahi-client as a dependency. 
Additionally, the ot-br-posix version was bumped to include the latest commits made since I began working on this.

wdyt?

**Comment by neheb** [comment]
Meh I bought 2 USB sticks to test with this. Never got around to it...

**Comment by stintel** [comment]
Could you fix the test formalities failure though?
> : Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: Itay Shoshani <itai.sho@gmail.com>')

File: net/openthread-br/Makefile [hunk 1]
```diff
@@ -41,7 +41,8 @@ define Package/openthread-br
 	+libubox \
 	+libubus \
 	+mdnsd \
-	+mdnsresponder
+	+mdnsresponder \
+	+kmod-tun
```
Comment (stintel) [inline net/openthread-br/Makefile:45]: Please maintain alphabetical order

File: net/openthread-br/Makefile [hunk 2]
```diff
@@ -67,7 +70,8 @@ CMAKE_OPTIONS += \
 	-DOTBR_REST=ON \
 	-DOTBR_SRP_ADVERTISING_PROXY=ON \
 	-DOTBR_SRP_SERVER_AUTO_ENABLE=ON \
-	-DOTBR_TREL=ON
+	-DOTBR_TREL=ON \
+	-DOTBR_NAT64=OFF
```
Comment (stintel) [inline net/openthread-br/Makefile:74]: Same here, maintain alphabetical order.

**Comment by stintel** [comment]
And maybe add some of the reasoning you mentioned in the comments for changes in the commit message.

**Comment by itay-sho** [comment]
Worked on the changes you talked about
Can you take a look?

**Comment by stintel** [comment]
Had guests over the weekend, but just tested this and can't spot any immediate regressions.

**Pull request merged**